merkletree.go

```diff
@@ -659,106 +648,18 @@ func CircomSiblingsFromSiblings(siblings []*Hash, levels int) []*Hash {
 	return siblings
 }
 
-// CircomProcessorProof defines the ProcessorProof compatible with circom. Is
-// the data of the proof between the transition from one state to another.
-type CircomProcessorProof struct {
-	OldRoot  *Hash   `json:"oldRoot"`
-	NewRoot  *Hash   `json:"newRoot"`
-	Siblings []*Hash `json:"siblings"`
-	OldKey   *Hash   `json:"oldKey"`
-	OldValue *Hash   `json:"oldValue"`
-	NewKey   *Hash   `json:"newKey"`
-	NewValue *Hash   `json:"newValue"`
-	IsOld0   bool    `json:"isOld0"`
+// TransactionInfo defines information about change merkletree.
+type TransactionInfo struct {
```

Comment:
I think we should keep this code on our side. Because the user may make a mistake when filling out the structure. I also think it is good to know how the tree changed after each operation.

Reply:
The MT transition info is something clients will need in the near future according to info from @demonsh and @OBrezhniev .
So it is better to return it after the update.
@olomix, yes, a client may call the state before & after but then we need to solve concurrent client calls in some way, which leads us to additional "umbrella" method implementation which result to even more complexity that it is now.

Reply:
@olomix could I you resolve this conversation?